### ISSUE TITLE
How to change the font of the text in qtsasstheme?

### ISSUE THREAD

**Issue by Liskelleo** [yjg30737/pyqt-light-gray-theme#1]
Sorry for making this issue such a mess because of my mistouching the buttons, this issue should have been in the issue section of "qt-sass-theme"... Anyway, my questions are:
1) How to change the font of the text in qt-sass-theme (I cannot figure out those `.scss` files)?
2) How to make the color of the statusbar's text to white when the theme is set as 'dark-grey' (it seems to be an issue)?
Looking forward to your reply^^

**Comment by yjg30737** [comment]
#1 Well.. I've actually added it but i did this such a long time ago so i couldn't figure out how to do it (so far)

I will mange to set the font size by qtsasstheme only.

But now, i can't do it because i'm kinda busy right now, so instead i give you this "temporary" way

```python
app = QApplication(sys.argv)
widget = MainWindow()
app.setFont(QFont('Arial', 14)) # This is how you do it.. for now
...
```

#2 I've just added QStatusBar to work just like that.

I didn't upload it to pip so you have to clone it to test it out, i will upload it 12 hours later or something for personal reason

Don't hesistate to ask if you have any questions about this 🙂

**Comment by Liskelleo** [comment]
> this "temporary" way

This method doesn't seem to work for me when I tried to switch between different themes. This method seems to only set an initial style, which means that it doesn't work when I change the style by using the slot function of a button or an action.

**Comment by yjg30737** [comment]
I don't know how you did that but that's quite astonishing that you actually attempted to switch the theme on runtime. I didn't expect that since i've never made this for that purpose, or at least as i remember.

Did theme switching actually work on runtime? That's cool.

And yes, that is temporary way. I added the "fontsize" variable to .scss file to make user able to dynamically change the font size. I'm still working on it, by the way. It will take a while.

**Comment by Liskelleo** [comment]
> added the "fontsize" variable

Great. Any progress or new methods to use if you make you can tell me^^

>Did theme switching actually work on runtime?

Try this code:
```
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtCore import Qt, QObject
from PyQt5.QtWidgets import QMainWindow, QApplication, QMenu, QMenuBar, QAction
from qt_sass_theme.qtSassTheme import QtSassTheme
from pyqt_hidable_menubar import HidableMenuBar

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(800, 600)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        MainWindow.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))

class MyMainWin(QMainWindow, Ui_MainWindow):
    def __init__(self):
        super(MyMainWin, self).__init__() 
        self.setupUi(self)
        self.show()
        self.initMenu()

    def initMenu(self):
        hidableMenuBar = HidableMenuBar()
        self.filemenu3 = QMenu("Configure(&C)", self)

        self.filemenu3_2 = QMenu("Theme", self)
        self.dark_action = QAction(self)
        self.dark_action.setCheckable(False)
        self.dark_action.setObjectName('darkAction')
        self.dark_action.triggered.connect(self.__themeToggled1)
        self.dark_action.setText('Dark')
        self.dark_action.setCheckable(True)
        self.dark_action.setChecked(False)

        self.darkblue_action = QAction(self)
        self.darkblue_action.setCheckable(False)
        self.darkblue_action.setObjectName('darkblueAction')
        self.darkblue_action.triggered.connect(self.__themeToggled2)
        self.darkblue_action.setText('DeepBlue')
        self.darkblue_action.setCheckable(True)
        self.darkblue_action.setChecked(False)

        self.blue_action = QAction(self)
        self.blue_action.setCheckable(False)
        self.blue_action.setObjectName('blueAction')
        self.blue_action.triggered.connect(self.__themeToggled3)
        self.blue_action.setText('DarkBlue')
        self.blue_action.setCheckable(True)
        self.blue_action.setChecked(False)

        self.filemenu3.addMenu(self.filemenu3_2)
        self.filemenu3_2.addAction(self.dark_action)
        self.filemenu3_2.addAction(self.darkblue_action)
        self.filemenu3_2.addAction(self.blue_action)
        hidableMenuBar.addMenu(self.filemenu3)
        self.setMenuBar(hidableMenuBar)

    def __themeToggled1(self, f):
        if self.dark_action.isChecked():
            t = QtSassTheme()
            t.getThemeFiles(theme='dark_gray')
            t.setThemeFiles(main_window=self)
            self.darkblue_action.setChecked(False)
            self.blue_action.setChecked(False)
        else:
            t = QtSassTheme()
            t.getThemeFiles(theme='light_gray')
            t.setThemeFiles(main_window=self)

    def __themeToggled2(self, f):
        if self.darkblue_action.isChecked():
            t = QtSassTheme()
            t.getThemeFiles(theme='dark_blue')
            t.setThemeFiles(main_window=self)
            self.dark_action.setChecked(False)
            self.blue_action.setChecked(False)
        else:
            t = QtSassTheme()
            t.getThemeFiles(theme='light_gray')
            t.setThemeFiles(main_window=self)
    
    def __themeToggled3(self, f):
        if self.blue_action.isChecked():
            t = QtSassTheme()
            t.getThemeFiles(theme='light_blue')
            t.setThemeFiles(main_window=self)
            self.dark_action.setChecked(False)
            self.darkblue_action.setChecked(False)
        else:
            t = QtSassTheme()
            t.getThemeFiles(theme='light_gray')
            t.setThemeFiles(main_window=self)

if __name__ == "__main__":
    import sys
    QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)
    app = QApplication(sys.argv)
    main = MyMainWin()
    main.show()
    sys.exit(app.exec_())
```
I note that (on Win10) if I delete the code "QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)" then menu actions on hidable menubar will "disappear" when I switch the theme or change the visibility of hidable menubar, and to say "disappear" I mean the menu could only be open like this:
![image](https://user-images.githubusercontent.com/97718634/210465148-6bd0fafd-a537-49b7-94c7-07cd0b05c2d7.png)
I don't know why this happens. Please tell me the reason if you know, and I would appreciate for that.

**Comment by yjg30737** [comment]
The code you gave me is very astonishing. 👍

I think i might know what you intend to do.

Fully-responsive widgets based on font size.

That happened because menu bar's height is small compared to the font size(height, in this case). If menu text is bigger than menu bar, menu text puts itself into the side-menu to keep the UI intact.

If all widgets (including the menu bar) is fully-responsive based on text they have, that will be solved.

**Comment by Liskelleo** [comment]
Thank you, what you think about my work is right^^
And if you finish your work on the "fontsize" part, don't forget to let me know^^

**Comment by yjg30737** [comment]
First of all, i've found that "standard" widgets of Qt are somewhat responsive(suprisingly) already not only Windows 11 but also Windows 10 as well.

I've tested it out at most basic widgets such as QLabel and QPushButton and every font and size of the widget containing that font changed at once.

But `HideMenuBar` you used is customized widget and i set the animation of it with setFixedHeight (there was no other choice), so it is completely unresponsive. After all, the height is fixed.

In this case, i have to write this code:

```python
    def event(self, e):
        if int(e.type()) == 76:
            height = self.sizeHint().height()
            self.__menuAnimation.setStartValue(height)
        return super().event(e)
```

As far as i concern, <a href="https://doc.qt.io/qt-6/qevent.html">LayoutRequest(Value is 76)</a> is most generic event to handle when app got resized. FontChange and StyleChange event are not enough to handle that issue.

I've just updated `HideMenuBar` like that and it should be responsive from now on.

Second, i add the fontsize and that works like a charm.

For example(based on your previous example):

```python
t = QtSassTheme()
t.getThemeFiles(theme='dark_gray', font_size=15)
```

**Comment by Liskelleo** [comment]
Thank you for helping me with all these things^^I really appreciate that^^

**Comment by Liskelleo** [comment]
Sorry to reopen this issue, because there seems to be another issue after setting the "global" fontsize of the QMainWindow by your extension (from my perspective), the fontszie in my TextEdit cannot be modified by QSlider, like this: (QAQ)
![image](https://user-images.githubusercontent.com/97718634/210963093-5765ade6-5cff-4f1d-9ad2-00dc5d63d79f.png)
So how to handle this? I'm now trying to work on it and btw, my TextEdit is based on your DarkNotePadTextEdit, the code related to it is shown below:
```
        self.statusBar = QStatusBar()
        self.setStatusBar(self.statusBar)
        self.statusBar.setSizeGripEnabled(False)

        self.__zoomScaleText = 'fontsize: {0}% '
        self.__charsLinesCountText = 'char: {0}'
        self.__charsLinesCountLabel = QLabel('char: 0')

        text = self.__zoomScaleText.format(self.textEdit.getScale())
        self.__zoomScaleLabel = QLabel()
        self.__zoomScaleLabel.setText(text)

        self.__zoomScaleSlider = QSlider()
        self.__zoomScaleSlider.setOrientation(Qt.Horizontal)
        self.__zoomScaleSlider.setRange(50, 250)
        self.__zoomScaleSlider.setValue(self.textEdit.getScale())
        self.__zoomScaleSlider.valueChanged.connect(self.__zoomScaleSliderValueChanged)

        self.__zoomScaleSlider.setTickInterval(10)
        self.__zoomScaleSlider.setSingleStep(10)
        self.__zoomScaleSlider.setMaximumWidth(self.__zoomScaleSlider.sizeHint().width())

        self.statusBar.addPermanentWidget(self.__zoomScaleLabel)
        self.statusBar.addPermanentWidget(self.__zoomScaleSlider)
        self.statusBar.addPermanentWidget(self.__charsLinesCountLabel)
```
......
```
    def __zoomByWheel(self, n):
        self.__zoomScaleLabel.setText(self.__zoomScaleText.format(n))
        self.__zoomScaleSlider.setValue(n)

    def __zoomIn(self):
        self.textEdit.zoomIn(10)
        self.__zoomScaleLabel.setText(self.__zoomScaleText.format(self.textEdit.getScale()))
        self.__zoomScaleSlider.setValue(self.textEdit.getScale())

    def __zoomOut(self):
        self.textEdit.zoomOut(10)
        self.__zoomScaleLabel.setText(self.__zoomScaleText.format(self.textEdit.getScale()))
        self.__zoomScaleSlider.setValue(self.textEdit.getScale())

    def __zoomReset(self):
        self.textEdit.zoomInit()
        self.__zoomScaleLabel.setText(self.__zoomScaleText.format(100))
        self.__zoomScaleSlider.setValue(self.textEdit.getScale())

    def __zoomScaleSliderValueChanged(self, v):
        v = v - v % 10
        self.textEdit.setScale(v)
        text = self.__zoomScaleText.format(self.textEdit.getScale())
        self.__zoomScaleLabel.setText(text)
        self.__zoomScaleLabel.setMaximumWidth(self.__zoomScaleLabel.fontMetrics().boundingRect(text).width()+5)
```
And I've noticed that the layout of the hidable menubar changed slightly after switching between different themes, and the size of the close button changed slightly after changing the visibility of the menubar, please look CAREFULLY at the two groups of pictures I captured below: 
(a)before and after switching between different themes
![image](https://user-images.githubusercontent.com/97718634/210969836-74e2addb-8fdd-4135-8a5c-46055461dfab.png)
![image](https://user-images.githubusercontent.com/97718634/210969857-684663c7-f7b3-4e5f-8cae-aa1ef5629a7c.png)
(b)before and after changing the visibility of the menubar
![image](https://user-images.githubusercontent.com/97718634/210969665-25298181-54c4-4b11-8aee-8589fca2c42b.png)
![image](https://user-images.githubusercontent.com/97718634/210969668-c0b96d70-bfaf-4f57-b5c1-55333c75fd41.png)
The latter issue might not be a bother since I've noticed that before I opened this issue section and I'm sorry for bothering so much, but I'm still looking forward to your reply^^

**Comment by yjg30737** [comment]
I've just solved that problem by using QApplication.setFont instead of using CSS to set font size.

Button maintains its own size as well.

TL;DR It will work fine.

Here is the lesson: Don't use CSS to set the font size or something like that. That will fix the font size permanantly.

The reason why they did that is beyond me.

If you have any questions, feel free to ask.

**Comment by Liskelleo** [comment]
It works fine to me^^ Thank you!
Hmm, however, (please forgive me, I’m just a bit OCD-ish) the font size between the menu bar and menu actions are slightly out of balance, and the same scenario happens with some gizmos, like QMesseageBox, QDialougue. Look the pictures I captured below:

(a) what the menu looks like after switching the theme 
![image](https://user-images.githubusercontent.com/97718634/211258164-8430e18f-e92e-4a84-847d-500736707896.png)

(bb) before and after changing the default theme to dark theme
![image](https://user-images.githubusercontent.com/97718634/211259418-0fd62a48-0ec5-4ae4-b9bd-606e4120b40d.png)
![image](https://user-images.githubusercontent.com/97718634/211259421-3548e5d1-811f-4340-b338-67467e5c9f2a.png)

Anyway, this issue might still not be a problem and I'm sorry for bothering.
Looking forward to your reply^^

**Comment by yjg30737** [comment]
It may work now.

I had set the range of the font for some reasons in that function, i've just remove that all.

**Comment by Liskelleo** [comment]
OK, then. It now works perfectly for me!
Thanks for all you've done^^
